### PR TITLE
Add up-to-date ETW for WriteLinesToFile

### PR DESCRIPTION
**Pull request by Forgind**
I looked through a trace of which tasks ran even for the second identical build. Many didn't have a "taskName," that is, _taskNode or _taskNode.Name was null, which surprised me. Of the rest, none in the MSBuild repo had up-to-date check behavior, as far as I could tell, except this one, so I added a check here. If I am correct, this finishes #6658

### CODE REVIEW

File: src/Tasks/FileIO/WriteLinesToFile.cs [hunk 1]
```diff
@@ -87,6 +88,7 @@ public override bool Execute()
                     {
                         Directory.CreateDirectory(directoryPath);
                         string contentsAsString = buffer.ToString();
+                        MSBuildEventSource.Log.WriteLinesToFileUpToDateStart();
```
Comment (rainersigwald) [inline src/Tasks/FileIO/WriteLinesToFile.cs:91]: It makes me uncomfortable to emit this in (the standard) cases where `WriteOnlyWhenDifferent` is unset/false. Should we condition it?

Comment (Forgind) [reply src/Tasks/FileIO/WriteLinesToFile.cs:91]: It's implicitly conditioned on events being enabled, so it won't be firing tons of traces into the void. Is it just that you want to be clear whether our check is super fast versus only checking WriteOnlyWhenDifferent?

Comment (rainersigwald) [reply src/Tasks/FileIO/WriteLinesToFile.cs:91]: Yes, I'd like to be able to distinguish when we even tried to do the up-to-date check.

File: src/Framework/MSBuildEventSource.cs [hunk 2]
```diff
@@ -423,6 +423,18 @@ public void PacketReadSize(int size)
...
         }
+
+        [Event(60, Keywords = Keywords.All)]
+        public void WriteLinesToFileUpToDateStart()
+        {
+            WriteEvent(60);
+        }
+
+        [Event(61, Keywords = Keywords.All)]
+        public void WriteLinesToFileUpToDateStop(string fileItemSpec)
```
Comment (rainersigwald) [inline src/Framework/MSBuildEventSource.cs:434]:
```suggestion
        public void WriteLinesToFileUpToDateStop(string fileItemSpec, bool wasUpToDate)
```

File: src/Tasks/FileIO/WriteLinesToFile.cs [hunk 3]
```diff
@@ -87,27 +88,32 @@ public override bool Execute()
...
-                                    if (existingContents.Equals(contentsAsString))
+                                    string existingContents = System.IO.File.ReadAllText(File.ItemSpec);
+                                    if (existingContents.Length == buffer.Length)
                                     {
-                                        Log.LogMessageFromResources(MessageImportance.Low, "WriteLinesToFile.SkippingUnchangedFile", File.ItemSpec);
-                                        return true;
+                                        if (existingContents.Equals(contentsAsString))
+                                        {
+                                            Log.LogMessageFromResources(MessageImportance.Low, "WriteLinesToFile.SkippingUnchangedFile", File.ItemSpec);
+                                            return true;
```
Comment (rainersigwald) [inline src/Tasks/FileIO/WriteLinesToFile.cs:106]:
```suggestion
                                            MSBuildEventSource.Log.WriteLinesToFileUpToDateStop(File.ItemSpec, true);
                                            return true;
```

File: src/Tasks/FileIO/WriteLinesToFile.cs [hunk 4]
```diff
@@ -87,27 +88,32 @@ public override bool Execute()
...
+                                        }
                                     }
                                 }
                             }
+                            catch (IOException)
+                            {
+                                Log.LogMessageFromResources(MessageImportance.Low, "WriteLinesToFile.ErrorReadingFile", File.ItemSpec);
+                            }
+                            MSBuildEventSource.Log.WriteLinesToFileUpToDateStop(File.ItemSpec);
```
Comment (rainersigwald) [inline src/Tasks/FileIO/WriteLinesToFile.cs:115]:
```suggestion
                            MSBuildEventSource.Log.WriteLinesToFileUpToDateStop(File.ItemSpec, false);
```

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**